File: edukate-backend/src/main/java/io/github/sanyavertolet/edukate/backend/dtos/FileMetadata.java
package io.github.sanyavertolet.edukate.backend.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.With;

import java.time.Instant;

@Data
@AllArgsConstructor
public class FileMetadata {
    @With
    private String key;
    private String authorName;
    private Instant lastModified;
    private Long size;

    public static FileMetadata of(String key, String authorName, Instant lastModified, Long size) {
        return new FileMetadata(key, authorName, lastModified, size);
    }
}

File: edukate-backend/src/main/kotlin/io/github/sanyavertolet/edukate/backend/dtos/FileMetadata.kt
package io.github.sanyavertolet.edukate.backend.dtos

import java.time.Instant

data class FileMetadata(
    val key: String,
    val authorName: String,
    val lastModified: Instant,
    val size: Long,
) {
    companion object {
        @JvmStatic
        fun of(
            key: String, authorName: String, lastModified: Instant, size: Long
        ) = FileMetadata(key, authorName, lastModified, size)
    }
}
